app/controllers/application_controller.rb

```diff
@@ -190,11 +191,24 @@ def expanded_embed(url, caption: nil, link: nil)
       end
     end
 
-    render_to_string partial: "/articles/embeds/#{slug}.html.erb", locals: { embed_id: embed_id || url, caption: caption, link: link }
+    render_to_string partial: "/articles/embeds/#{slug}.html.erb", locals: { embed_id: embed_id || url, caption: caption, link: link, id: id }
   end
   helper_method :expanded_embed
 
   def url_or_path?(string)
     string.match?(/^(http|\/)\S+/)
   end
+
+  def remove_id(pieces)
+    return unless pieces.present?
+
+    id_string = pieces.detect { |piece| piece =~ /id:\S+/ }
+
+    if id_string
+      id = id_string.split(":").last
+      pieces.delete(id_string)
```

Comment:
pieces isn't a local var, it's passed by reference as an argument.  You can see a simpler version here:

```ruby
>> def remove_first(array)
>>   array.shift
>> end

>> test_array = [1, 2, 3]
=> [1, 2, 3]

>> remove_first(test_array)
=> 1

>> test_array
=> [2, 3]
```